### PR TITLE
Alexander Lee's Pull Request

### PR DESCRIPTION
**Pull request by shadow2418889**
## Submission Checklist:

- [x] I am a current high school, middle school, or home schooled student.
  - [x] I have filled out the [verification form](https://airtable.com/app4Bs8Tjwvk5qcD4/pagxECjJZOgvKVnLd/form)
- [x] I have joined the [`#onboard`](https://hackclub.slack.com/archives/C056AMWSFKJ) channel on Slack
- [x] I made this board from scratch, even if I followed a tutorial
- [x] I have followed the directions in [README.md](https://github.com/hackclub/OnBoard/blob/main/README.md#getting-started)
  - [x] Created a folder under `onboard/projects`
  - [x] Filled out `TEMPLATE.md` as `README.md`
  - [x] Uploaded Gerber, sources, and schematic
  - [x] Uploaded screenshot of PCB order with all costs included
  - [x] I am ordering the smallest batch size (ie. JLCPCB's smallest order is for 5 boards)
  - [n/a] If outside the US I've checked that I can afford the customs charges in my country (which isn't covered by the grant)

- [ ] (Optional) This project is from a tutorial[^1].
- [x] (Optional) I'm in a FIRST (FRC, FTC, FLL, etc.) team. The number is: 125

[^1]: Projects from a tutorial are 100% fine! We just want to ask so we can count how people are using tutorials.

### CODE REVIEW

**Comment by prophetorpheus** [comment]
Hi, I'm Orpheus Leap! Here to help you review your PR.

  You can view a render of your board over on [tracespace.io](<https://tracespace.io/view/?boardUrl=https://raw.githubusercontent.com/hackclub/OnBoard/d590f2999cb9d0e4a6f096a6c6553586fd91806c/projects/Alexander's ESP32 Omini Directional PCB Robot/Gerber_PCB_Robot Tour 2023 V0.1.zip>)!

  Happy OnBoarding!

**Review by karmanyaahm — CHANGES_REQUESTED** [review]
LOVE THIS IDEA!!!!

Small blocker, just like your upload your PCB_PCB.json file, please File > Export > EasyEDA a JSON  from the Schematic tab of your project.


This is not a requirement, but there are definitely ways to optimize your design to fit it in <$100; I don't know whether you can achieve that with your competition timeline.

**Cost saving suggestions** (there are more, these are just the few things that stood out to me, ask questions on Slack):
Just a suggestion though, if you replace the ESP32 module with an ESP32-SOLO-1, you might be able to save a lot of money. Right now your ESP32 S2 module is forcing you to use Standard Assembly, which adds like $20 to your assembly cost. I don't know how practical this is in your application, so this is just a suggestion that you may or may not want to take.

Here are also a few notes to get basically free money:
1. These few capacitors are all "Extended parts". This means that it costs $3 extra per part for JLCPCB to grab these.
image In your schematic, you can replace these with generic capacitors with an equivalent value, and they should work the same. 
image
2. Same thing applies to these transistors. 
image You can go to https://jlcpcb.com/parts. Search "transistor", and select the "Basic parts" checkbox.
3. And buttons are the same as transistors
image






cc @Hugoyhu review circuit things plz?

**Review by sarthaktexas — APPROVED** [review]
Howdy Alex! I'm approving your student verification :)

**Comment by karmanyaahm** [comment]
@shadow2418889 lmk if there's any way I can help you get your grant sooner!

**Comment by Hugoyhu** [comment]
HI! Sorry for the late reply. @shadow2418889 let me know if you'd like a schematic / routing review.

**Comment by shadow2418889** [comment]
@Hugoyhu A review would be greatly appreciated! Any tips on how I could improve my design for the future would also be nice. Thank you all, Alexander

**Comment by Hugoyhu** [comment]
@shadow2418889 can you upload your schematic .json to the /src folder? Thank you

**Comment by shadow2418889** [comment]
I uploaded the export from EasyEda. Thank you all again.

**Review by Hugoyhu — CHANGES_REQUESTED** [review]
Hi all and sorry for the wait. Currently checking your schematic. Few things:

- On U2 (CP2102N-28) why is RXD and TXD swapped? I'm referencing [the Adafruit CP2102N-24 Friend.](https://cdn-learn.adafruit.com/assets/assets/000/111/711/original/adafruit_products_schem.png?1652465596)
- Should VREGIN be connected to 3.3V? [SiLabs datasheet says 3.6-5,25 per fig 2.2](https://www.silabs.com/documents/public/data-sheets/cp2102n-datasheet.pdf)

I will take a look at the other portions of this sch soon, thse two are issues taht pop out to me for the USB-serial portion.

**Comment by shadow2418889** [comment]
Hi Hugo,
First of all, thank you for your help and I'm completely fine about the delay.

For RXD and TXD swappage I was just following the schematics the ESPRESSIF M-1 Devkit:
image
and Feranec's tutorial https://youtu.be/S_p0YV-JlfU?t=2161. Honestly, I have no idea why the inputs are swapped, but I trust the schematic and Feranec. 

The VREGIN connection also comes from the ESPRESSIF M-1 Devkit
image
While I am outside the recommended voltage, the datasheet states that the CP2102N can handle an input voltage 3V, so uh, I guess I can pray that it works out fine.

Thanks, Alexander

File: projects/Alexander's ESP32 Omini Directional PCB Robot/Schematic_Robot Tour 2023_2024-01-28.pdf
Comment (Hugoyhu) [inline projects/Alexander's ESP32 Omini Directional PCB Robot/Schematic_Robot Tour 2023_2024-01-28.pdf:1]: Currently reading the datasheet for TPS62745 [here](https://www.ti.com/lit/ds/symlink/tps62745.pdf?ts=1707383667608&ref_url=https%253A%252F%252Fwww.google.com%252F): it seems your VSEL pins are connected to VOUT and not GND.

There's also a newer part TI recommends with "smaller size, better efficiency" which might be ideal for your battery-powered project: [TPS629203](https://www.ti.com/product/TPS629203)

I'll take a looka t your motor drivers next (nice part selection! Fun coincidence, we were looking at these for our Blot machine). I'll make a comment when I am done checking that part.

For the previous comment RE CP2102N and etc: I have checked against the Espressif schematic and it seems OK. If you followed that then it should be good and I don't immediately see any problems.

**Review by karmanyaahm — APPROVED** [review]
I'm approving this for now. Please look at Hugo's comments to make sure your robot will work. Your grant should be sent soon (ping @sarthaktexas)!

**Comment by shadow2418889** [comment]
Thank you all! I finished the changes and I'll talk to Sarthak.

File: projects/Alexander's ESP32 Omini Directional PCB Robot/Schematic_Robot Tour 2023_2024-01-28.pdf
Comment (shadow2418889) [reply projects/Alexander's ESP32 Omini Directional PCB Robot/Schematic_Robot Tour 2023_2024-01-28.pdf:1]: I'm looking for an output of 3.3v, so I just followed the recommended schematics:
image
JLCPCB doesn't have any TPS629203 in stock unfortunately.

**Review by Hugoyhu — COMMENTED** [review]
This seems more or less OK for the schematic and well put together. I don't see any issues for DRV8833 against datasheet. Habve you run DRC on your PCB?

**Comment by Hugoyhu** [comment]
Oh, one note on PCB. Do you paln on using the BT / WiFi for ESP32? If so, the module should be on the edge of the PCB. at the very least, the area under the antenna section should have no copper fill, and at best, there should be a cutout or the antenna on module extends out.

**Comment by shadow2418889** [comment]
I was gonna play with Bluetooth control but it wasn't really vital for the project. Thanks for letting me know, ill probably update the design to better support wireless connectivity sometime in the future.

**Pull request merged**